libs/index/odc/index/stac.py

```diff
@@ -29,42 +29,64 @@ def _stac_product_lookup(item):
                 properties['sentinel:latitude_band'],
                 properties['sentinel:grid_square']
             )
+            default_grid = "g10m"
+    elif properties.get('platform') in LANDSAT_PLATFORMS:
+        product_label = _product_label(item)
+        product_name = properties.get('odc:product')
+        region_code = properties.get('odc:region_code')
+        default_grid = "g30m"
+    else:
+        product_label = item['id']
+        product_name = properties['platform']
```

Comment:
are these items guaranteed to be present?

Reply:
The `id` guaranteed to be present as it is REQUIRED field but `platform` is not REQUIRED field
https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#item-fields